website/docs/reference/warehouse-profiles/mssql-profile.md

```diff
@@ -97,106 +149,200 @@ your_profile_name:
 
 </TabItem>
 
-<TabItem value="cli">
+<TabItem value="service_principal">
+
+Client ID is often also referred to as Application ID.
+
+<File name='profiles.yml'>
+
+```yaml
+your_profile_name:
+  target: dev
+  outputs:
+    dev:
+      type: sqlserver
+      driver: 'ODBC Driver 17 for SQL Server' # (The ODBC Driver installed on your system)
+      server: hostname or IP of your server
+      port: 1433
+      schema: schema_name
+      authentication: ServicePrincipal
+      tenant_id: 00000000-0000-0000-0000-000000001234
+      client_id: 00000000-0000-0000-0000-000000001234
+      client_secret: S3cret!
+```
+
+</File>
+
+</TabItem>
+
+<TabItem value="managed_identity">
+
+Both system-assigned and user-assigned managed identities will work.
+
+<File name='profiles.yml'>
+
+```yaml
+your_profile_name:
+  target: dev
+  outputs:
+    dev:
+      type: sqlserver
+      driver: 'ODBC Driver 17 for SQL Server' # (The ODBC Driver installed on your system)
+      server: hostname or IP of your server
+      port: 1433
+      schema: schema_name
+      authentication: MSI
+```
+
+</File>
+
+</TabItem>
+
+<TabItem value="environment_based">
+
+This authentication option allows you to dynamically select an authentication method depending on the available environment variables.
+
+[The Microsoft docs on EnvironmentCredential](https://docs.microsoft.com/en-us/python/api/azure-identity/azure.identity.environmentcredential?view=azure-python)
+explain the available combinations of environment variables you can use.
+
+<File name='profiles.yml'>
+
+```yaml
+your_profile_name:
+  target: dev
+  outputs:
+    dev:
+      type: sqlserver
+      driver: 'ODBC Driver 17 for SQL Server' # (The ODBC Driver installed on your system)
+      server: hostname or IP of your server
+      port: 1433
+      schema: schema_name
+      authentication: environment
+```
+
+</File>
+
+</TabItem>
+
+<TabItem value="azure_cli">
 
 First, install the [Azure CLI](https://docs.microsoft.com/en-us/cli/azure/install-azure-cli), then, log in:
 
 `az login`
 
 <File name='profiles.yml'>
 
-```yml
+```yaml
 your_profile_name:
   target: dev
   outputs:
     dev:
       type: sqlserver
-      driver: 'ODBC Driver 17 for SQL Server' (The ODBC Driver installed on your system)
-      server: server-host-name or ip
+      driver: 'ODBC Driver 17 for SQL Server' # (The ODBC Driver installed on your system)
+      server: hostname or IP of your server
       port: 1433
-      schema: schemaname
+      schema: schema_name
       authentication: CLI
 ```
-This is also the preferred route for using a service principal:
-
-`az login --service-principal --username $CLIENTID --password $SECRET --tenant $TENANTID`
 
 </File>
 
 </TabItem>
 
-<TabItem value="interactive">
+<TabItem value="auto">
```

Comment:
I didn't even know this existed?!? lol

Reply:
This the one I'd recommend most. Just login with anything and it will work.